File: hazelcast-getting-started-eachlanguage/scala/src/main/scala/HazelcastGettingStarted.scala
// Execution Command: $ sbt run

import com.hazelcast.config.Config
import com.hazelcast.core.Hazelcast

object HazelcastGettingStarted {
  def main(args: Array[String]): Unit = {
    val cfg = new Config
    val instance = Hazelcast.newHazelcastInstance(cfg)

    val mapCustomers = instance.getMap[Int, String]("customers")

    for ((i, name) <- Array((1, "Joe"),
                            (2, "Ali"),
                            (3, "Avi"))) {
      mapCustomers.put(i, name)
    }

    println(s"Customer with key 1: ${mapCustomers.get(1)}")
    println(s"Map Size: ${mapCustomers.size}")


    val queueCustomers = instance.getQueue[String]("customers")

    Array("Tom", "Mary", "Jane").foreach(queueCustomers.offer)

    println(s"First Customer: ${queueCustomers.poll()}")
    println(s"Second Customer: ${queueCustomers.peek}")
    println(s"Queue size: ${queueCustomers.size}")

    instance.getLifecycleService.shutdown()
  }
}

File: hazelcast-getting-started-eachlanguage/clojure/hazelcast-getting-started.clj
;; Execution Command: $ lein exec hazelcast-getting-started.clj

(require '[leiningen.exec :as exec])

(exec/deps '[[com.hazelcast/hazelcast "3.2"]])

(import '(com.hazelcast.config Config)
        '(com.hazelcast.core Hazelcast))

(let [cfg (Config.)
      instance (Hazelcast/newHazelcastInstance cfg)]

  (let [map-customers (. instance getMap "customers")]
    (doseq [[i name] '([1 "Joe"] [2 "Ali"] [3 "Avi"])]
      (. map-customers put i name))
    (println (str "Customer with key 1: " (. map-customers get 1)))
    (println (str "Map Size: " (. map-customers size))))

  (let [queue-customers (. instance getQueue "customers")]
    (doseq [name ["Tom" "Mary" "Jane"]]
      (. queue-customers offer name))
    (println (str "First customer: " (. queue-customers poll)))
    (println (str "Second customer: " (. queue-customers peek)))
    (println (str "Queue size: " (. queue-customers size))))

  (.. instance getLifecycleService shutdown))